Clang still has issues with not_null

Only in one place though, and that's in the test. I'm not sure whether we really care, if we really need to call `push_back` on a non-rvalue `not_null` it's easily circumvented. Expanded example from #720: http://goo.gl/4k9zFl.
The added code is
```c++
  not_null<T*> not_null_access_t;
  access_t = not_null_access_t;
  v.push_back(access_t);  // Works
  v.push_back(not_null_access_t);  // Fails with clang.
```
If we decide not to care about it, let's just comment out the `push_back` that fails on clang with an explanation, and we'll investigate if we ever have a reason to care again.

Error message:
```
base/not_null_test.cpp:161:16: error: no viable conversion from 'not_null<int *>' to 'value_type' (aka 'int *')
  v2.push_back(not_null_int);
               ^~~~~~~~~~~~
./base/not_null.hpp:196:3: note: candidate function not viable: no known conversion from 'not_null<[...]>' to 'not_null<[...]>' for object argument
  operator pointer&&() &&;
  ^
./base/not_null.hpp:193:3: note: candidate function
  operator pointer const&&() const&;
  ^
./base/not_null.hpp:229:3: note: candidate function
  operator bool() const;
  ^
/usr/include/c++/v1/vector:703:59: note: passing argument to parameter '__x' here
    _LIBCPP_INLINE_VISIBILITY void push_back(value_type&& __x);
                                                          ^
```

--- a/base/not_null_test.cpp
+++ b/base/not_null_test.cpp
@@ -5,6 +5,7 @@
 #include <string>
 #include <vector>
 
+#include "base/macros.hpp"
 #include "gtest/gtest.h"
 #include "gmock/gmock.h"
 
@@ -158,7 +159,20 @@ TEST_F(NotNullTest, RValue) {
   EXPECT_EQ(4, *not_null_owner_int);
 
   std::vector<int*> v2;
+  // NOTE(egg): The following fails using clang, I'm not sure where the bug is.
+  // More generally, if functions |foo(int*&&)| and |foo(int* const&)| exist,
+  // |foo(non_rvalue_not_null)| fails to compile with clang ("no viable
+  // conversion"), but without the |foo(int*&&)| overload it compiles.
+  // This is easily circumvented using a temporary (whereas the ambiguity that
+  // would result from having an |operator pointer const&() const&| would
+  // entirely prevent conversion of |not_null<unique_ptr<T>>| to
+  // |unique_ptr<T>|), so we ignore it.
+#if PRINCIPIA_COMPILER_MSVC
   v2.push_back(not_null_int);
+#else
+  int* const temporary = not_null_int;
+  v2.push_back(temporary);
+#endif
   EXPECT_EQ(2, *v2[0]);
   EXPECT_EQ(2, *not_null_int);
 

